src/Altinn.App.Api/Controllers/InstancesController.cs

```diff
@@ -1024,6 +1024,91 @@ await _dataClient.InsertFormData(
                 await UpdateDataValuesOnInstance(application.DataFields, targetInstance, dt.Id, data);
             }
         }
+
+        if (application.CopyInstanceSettings?.IncludeAttachments is null or false)
+        {
+            return;
+        }
+
+        // Copy binary data elements (files/attachments)
+        List<DataType> binaryDataTypes = application
+            .DataTypes.Where(dt => dt.AppLogic?.ClassRef == null)
+            .Where(dt =>
+                dt.TaskId != null
+                && dt.TaskId.Equals(targetInstance.Process.CurrentTask.ElementId, StringComparison.Ordinal)
+            )
```

Comment:
Looks like you still filter based on taskId, but the PR description says you don't?